subprojects/parseq/src/main/java/com/linkedin/parseq/retry/RetryPolicy.java

```diff
@@ -74,4 +76,27 @@ static RetryPolicy attemptsAndDuration(int attempts, long duration, long backoff
         setBackoffPolicy(BackoffPolicy.constant(backoff)).
         build();
   }
+
+  /**
+   * Retry policy with configurable number of retries, limited total duration, and limited rate of the encompassing task.
+   *
+   * @param attempts Total number of attempts (the number of retries will be that minus 1).
+   * @param duration Total duration of the task. This includes both the original request and all potential retries.
+   * @param rateLimiter The limiter to control the rate of retries per second.
+   * @param backoff The constant delay (in milliseconds) between retry attempts.
+   */
+  static RetryPolicy attemptsAndDurationAndRate(int attempts, long duration, RateLimiter rateLimiter, long backoff) {
```

Comment:
Now you added a policy based on three criteria, but I am wondering: does "using rateLimiter" as the only criteria work?

Reply:
using a ratelimiter by itself would work, but it will be suboptimal for many scenarios. Also, the combination of all these policies make the retry process fairer for all types of requests: retries and original requests.